### PR TITLE
Implemented the API token authorization in the service.

### PR DESCRIPTION
**Pull request by Sachinbisht27**
[#400](https://github.com/DostEducation/RP_IVR_analytics/issues/400)

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)

### Short description of what this resolves:
**Describe your changes in detail**
Added functionality to receive the API token from request and compare it with the auth token saved in the config.
If the token is not valid discard the request and log the error in the cloud function.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
+"""Flask configuration."""
 import os
 
 ENVIRONMENT = os.environ.get("ENVIRONMENT", "development")
 
+# Development env specific
```
Comment (Satendra-SR) [inline config.py]: Let's remove this as we are getting a gist from the next line?

File: utils/__init__.py [hunk 2]
```diff
@@ -1 +1,2 @@
 from .loggingutils import *
+from .api import *
```
Comment (Satendra-SR) [inline utils/__init__.py]: Can we rename it to `auth.py` instead?

**Review by Satendra-SR — COMMENTED** [review]
One comments

File: config.py (hunk 1 above)
Comment (Sachinbisht27) [reply config.py]: Sure

File: utils/__init__.py (hunk 2 above)
Comment (Sachinbisht27) [reply utils/__init__.py]: Yes, updating it.

**Review by Satendra-SR — APPROVED** [review]

**Pull request merged**